### PR TITLE
Evaluate selectors for non-optional gates

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR builds upon #1110, adding evaluations for all of the 'non-optional' gates, which will allow them to be removed from the linearization (in an upcoming PR).

This PR replaces #1105 as part of a rebase to make reduce the scope of changes needed for berkeley.

### CODE REVIEW

File: book/src/specs/kimchi.md [hunk 1]
```diff
@@ -1969,6 +1969,14 @@ pub struct ProofEvaluations<Evals> {
     pub generic_selector: Evals,
     /// evaluation of the poseidon selector polynomial
     pub poseidon_selector: Evals,
+    /// evaluation of the EC addition selector polynomial
+    pub complete_add_selector: Evals,
+    /// evaluation of the EC variable base scalar multiplication selector polynomial
+    pub mul_selector: Evals,
```
Comment (rbonichon) [inline book/src/specs/kimchi.md:1975]: [minor] EC does not seem to be used anywhere else in the book. I suggest to expand it like so
```suggestion
    /// evaluation of the elliptic curve addition selector polynomial
    pub complete_add_selector: Evals,
    /// evaluation of the elliptic curve variable base scalar multiplication selector polynomial
    pub mul_selector: Evals,
```

File: kimchi/src/proof.rs [hunk 2]
```diff
@@ -76,6 +76,14 @@ pub struct ProofEvaluations<Evals> {
     pub generic_selector: Evals,
     /// evaluation of the poseidon selector polynomial
     pub poseidon_selector: Evals,
+    /// evaluation of the EC addition selector polynomial
+    pub complete_add_selector: Evals,
+    /// evaluation of the EC variable base scalar multiplication selector polynomial
+    pub mul_selector: Evals,
```
Comment (rbonichon) [inline kimchi/src/proof.rs:82]: For homogeneity with the documentation
```suggestion
    /// evaluation of the elliptic curve addition selector polynomial
    pub complete_add_selector: Evals,
    /// evaluation of the elliptic curve variable base scalar multiplication selector polynomial
    pub mul_selector: Evals,
```

**Review by rbonichon — APPROVED** [review]
LGTM

File: book/src/specs/kimchi.md [hunk 3]
```diff
@@ -1969,6 +1969,14 @@ pub struct ProofEvaluations<Evals> {
     pub generic_selector: Evals,
     /// evaluation of the poseidon selector polynomial
     pub poseidon_selector: Evals,
+    /// evaluation of the EC addition selector polynomial
+    pub complete_add_selector: Evals,
```
Comment (querolita) [inline book/src/specs/kimchi.md:1973]: So this means only the purely optional gates such as Rot64 will be part of the linearization process? Got it

**Review by querolita — APPROVED** [review]

File: book/src/specs/kimchi.md [hunk 4]
```diff
@@ -1969,6 +1969,14 @@ pub struct ProofEvaluations<Evals> {
     pub generic_selector: Evals,
     /// evaluation of the poseidon selector polynomial
     pub poseidon_selector: Evals,
+    /// evaluation of the EC addition selector polynomial
```
Comment (dannywillems) [inline book/src/specs/kimchi.md:1972]: The image has to be updated in the high level IP also.

**Comment by dannywillems** [comment]
I don't know if it is relevant to this PR, but on the branch, I don't see the selectors added to `src/circuits/expr.rs`, `evaluate` L674. I run `git grep -A 10 "GateType::Poseidon` to see where other selectors are added in the code.

**Comment by dannywillems** [comment]
> I don't know if it is relevant to this PR, but on the branch, I don't see the selectors added to `src/circuits/expr.rs`, `evaluate` L674. I run `git grep -A 10 "GateType::Poseidon` to see where other selectors are added in the code.

Ah, added in this PR: https://github.com/o1-labs/proof-systems/pull/1106/files

**Pull request merged**